draft-ietf-quic-transport.md

```diff
@@ -1414,42 +1414,99 @@ TODO: see issue #161
 period of time.  A QUIC connection, once established, can be terminated in one
-of three ways:
-
-1. Explicit Shutdown: An endpoint sends a CONNECTION_CLOSE frame to
-   terminate the connection.  An endpoint MAY use application-layer mechanisms
-   prior to a CONNECTION_CLOSE to indicate that the connection will soon be
-   terminated.  On termination of the active streams, a CONNECTION_CLOSE may be
-   sent.  If an endpoint sends a CONNECTION_CLOSE frame while unterminated
-   streams are active (no FIN bit or RST_STREAM frames have been sent or
-   received for one or more streams), then the peer must assume that the streams
-   were incomplete and were abnormally terminated.
-
-2. Implicit Shutdown: The default idle timeout is a required parameter in
-   connection negotiation.  The maximum is 10 minutes.  If there is no network
-   activity for the duration of the idle timeout, the connection is closed.  By
-   default a CONNECTION_CLOSE frame will be sent.  A silent close option can be
-   enabled when it is expensive to send an explicit close, such as mobile
-   networks that must wake up the radio.
-
-3. Stateless Reset: An endpoint that loses state can use this procedure to cause
-   the connection to terminate early, see {{stateless-reset}} for details.
-
-After receiving either a CONNECTION_CLOSE frame or a Public Reset, an
-endpoint MUST NOT send additional packets on that connection. After
-sending either a CONNECTION_CLOSE frame or a Public Reset packet,
-implementations MUST NOT send any non-closing packets on that
-connection. If additional packets are received after this time and
-before idle_timeout seconds has passed, implementations SHOULD respond
-to them by sending a CONNECTION_CLOSE (which MAY just be a duplicate
-of the previous CONNECTION_CLOSE packet) but MAY also send a Public
-Reset packet.  Implementations SHOULD throttle these responses, for
-instance by exponentially backing off the number of packets which are
-received before sending a response.  After this time, implementations
-SHOULD respond to unexpected packets with a Public Reset packet.
-
-
-## Stateless Reset {#stateless-reset}
+of four ways:
+
+* application close ({{application-close}})
+* idle timeout ({{idle-timeout}})
+* immediate close ({{immediate-close}})
+* stateless reset ({{stateless-reset}})
+
+
+### Draining Period {#draining}
+
+After a connection is closed for any reason, an endpoint might receive packets
+from its peer.  These packets might have been sent prior to receiving any close
+signal, or they might be retransmissions of packets for which acknowledgments
+were lost.
+
+The draining period persists for three times the maximum of minimum RTO
+(kMinRTOTimeout) or the round trip time (smoothed_rtt), see {{QUIC-RECOVERY}}
+for descriptions of these values.  During this period, new packets can be
+attributed to the correct connection and acknowledged, but the connection is no
+longer considered active and usable.
```

Comment:
nit: instead of "the connection is no longer considered active and usable", how about "no new data or retransmittable packets may be sent on the connection"? The allowable frames are elucidated below, so I'd be happy with just "no new data may be sent on the connection."

Reply:
I went with the latter.  I'm trying *really* hard not to say that anything is retransmittable, packets or frames.  I want to keep reinforcing the idea that it's the *contents* of frames (the data or signal) that is retransmitted.